### PR TITLE
Update antreainstall.status from antrea clusteroperator.status

### PR DESCRIPTION
**Pull request by ruicao93**
1. Update antreainstall.status from antrea clusteroperator.status.
2. Show operator version in status.
3. Optimize config reconciliation:
    - If thre are only status fields changes in cluster config or
      antreainstall object, skip config difference check.

Signed-off-by: Rui Cao <rcao@vmware.com>

### CODE REVIEW

File: pkg/controller/status/status_controller.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("status-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to status of cluster operator object
+	err = c.Watch(&source.Kind{Type: &configv1.ClusterOperator{}}, &handler.EnqueueRequestForObject{})
```
Comment (dantingl) [inline pkg/controller/status/status_controller.go:54]: I don't think it's good to monitor cluster operator resource to update CRD status..
Later if we are going to make antrea operator generic, we need to change the way later as well.
We can just add a 'ready' status if anything is expected, otherwise we can update errors directly.

File: pkg/controller/config/config_controller.go [hunk 2]
```diff
@@ -116,6 +117,12 @@ func (r *ReconcileConfig) Reconcile(request reconcile.Request) (reconcile.Result
 		log.Error(err, "failed to get Cluster Network CR")
 		return reconcile.Result{Requeue: true}, err
 	}
+	if request.Name == clusterConfig.Name && r.appliedClusterConfig != nil {
```
Comment (dantingl) [inline pkg/controller/config/config_controller.go:120]: How about make this change in a different patch

Comment (ruicao93) [reply pkg/controller/config/config_controller.go:120]: Sure. Will create a new PR.
